pkg/cli/subcommands/check/command.go

```diff
@@ -172,8 +175,25 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 				apiimpl.Module(),
 				authtokenimpl.Module(),
 				fx.Supply(context.Background()),
-				fx.Provide(tagger.NewTaggerParamsForCoreAgent),
-				taggerimpl.Module(),
+				dualTaggerfx.Module(tagger.DualParams{
+					UseRemote: func(c config.Component) bool {
+						return pkgconfigsetup.IsCLCRunner(c) && c.GetBool("clc_runner_remote_tagger_enabled")
+					},
+				}, tagger.Params{}, tagger.RemoteParams{
+					RemoteTarget: func(config.Component) (string, error) {
+						target, err := utils.GetClusterAgentEndpoint()
+						if err != nil {
+							return "", err
+						}
+						return strings.TrimPrefix(target, "https://"), nil
+					},
+					RemoteTokenFetcher: func(c config.Component) func() (string, error) {
+						return func() (string, error) {
+							return security.GetClusterAgentAuthToken(c)
+						}
+					},
+					RemoteFilter: taggerTypes.NewFilterBuilder().Exclude(taggerTypes.KubernetesPodUID).Build(taggerTypes.HighCardinality),
+				}),
```

Comment:
This could also reuse a generic helpers to get the params.

Reply:
Done https://github.com/DataDog/datadog-agent/pull/30584/commits/08b167024bdcbfc2cc110dfe0effdfdb95f7b665